examples/front.kk

```diff
@@ -0,0 +1,89 @@
+/*----------------------------------------------------------------------------
+   Copyright 2024, Koka-Community Authors
+
+   Licensed under the MIT License ("The License"). You may not
+   use this file except in compliance with the License. A copy of the License
+   can be found in the LICENSE file at the root of this distribution.
+----------------------------------------------------------------------------*/
+
+import html/html
+
+import html/layout
+
+// todo: this deserves a PR to go into std/core?
+fun do(action : () -> e a) : e a
+  action()
+
+struct cnode
+  swarm : string
+  ip : string
+  ncores : int
+  avail-ram : int
+  total-ram : int
+
+effect val cnodes-info : list<cnode>
+
+fun frontpage(): <pure,cnodes-info> component
+  val template = tmpl/basic(page-head = fn(layout) {
+    page-head(tmpl/basic)(layout)
+    link("stylesheet", "text/css", "/front.css")
+  })
+  val title = "The Front Page"
+
+  with super <- default()
+  with page <- named handler
```

Comment:
how realistic it is, to add syntactic sugar to Koka, for surface syntax to omit methods simply delegating to `super`, in defining `page` here?

i.e. is it simple or complex in drafting a PR to add this?

Reply:
Yeah there are a lot of considerations here, ordering, allowing multiple different effects to have the same operation names etc. To fully support OO would be quite a change to possibly the type system as well, and not just a simple desugaring. In general I try to keep the number of operation clauses small, and then any time I think I need a new operation, see if I can make the current operations more general and provide functions that call those general clauses and do something more specific. The current design actually could be simplified a lot by creating a struct to hold the individual components and just making the effect have an operation for getting and setting that struct held in a state variable.